public/app/plugins/datasource/jaeger/datasource.ts

```diff
@@ -309,3 +310,16 @@ const emptyTraceDataFrame = new MutableDataFrame({
     },
   },
 });
+
+export function addNodeGraphFramesToResponse(response: DataQueryResponse): DataQueryResponse {
+  if (!response.data || response.data.length === 0) {
+    return response;
+  }
+
+  // Convert the first frame to a DataFrame for node graph processing
+  const frame = toDataFrame(response.data[0]);
+  return {
+    ...response,
+    data: [...response.data, ...createNodeGraphFrames(frame)],
```

Comment:
It's probably ok but still want to mention about this https://github.com/grafana/grafana/pull/102452
destructuring response or response data directly might cause issues.

Reply:
That's a good suggestion. I'll update and use `arr.concat` as suggested in https://github.com/grafana/grafana/pull/102452.

Reply:
Fixed in https://github.com/grafana/grafana/pull/104029/files/d87bd5842f1479df647ac8d92c58de04a339a217..ec393eabaa67af1db441d857602e9efb199d8304